src/server/game/AI/ScriptedAI/ScriptedEscortAI.cpp

```diff
@@ -384,9 +366,41 @@ void npc_escortAI::OnPossess(bool apply)
 
 void npc_escortAI::AddWaypoint(uint32 id, float x, float y, float z, uint32 waitTime)
 {
-    Escort_Waypoint t(id, x, y, z, waitTime);
+    int32 entry = -int32(me->GetEntry());
 
-    WaypointList.push_back(t);
+    WaypointPathContainer::iterator itr = sWaypointMgr->GetWaypointPathContainer().find(entry);
+    if (itr != sWaypointMgr->GetWaypointPathContainer().end())
+    {
+        for (WaypointPath::const_iterator it = itr->second.begin(); it != itr->second.end(); ++it)
+        {
+            if ((*it)->id == id)
+            {
+                itr->second.erase(it);
+                break;
+            }
+        }
+    }
+
+    Trinity::NormalizeMapCoord(x);
+    Trinity::NormalizeMapCoord(y);
+
+    WaypointPath& path = sWaypointMgr->GetWaypointPathContainer()[entry];
```

Comment:
Actually comments like that lead to the usage to automatic CI integration like Travis and AppVeyor that ensure PRs build on nix and Windows and don't add any warning. There is also a branch called tests https://github.com/TrinityCore/TrinityCore/tree/tests that would require (in a long far away future) all PRs to write unit tests to automatically test the code that could be run in helgrind too.

It seems lately there's an hurry to call a comment "useless" instead of spending some time understanding it and getting something valuable out of it.